cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaConnectConfigurationBuilderTest.java

```diff
@@ -514,4 +530,202 @@ public void withPluginPath() {
+        ));
+    }
+    
+    @ParallelTest
+    public void testStrimziMetricsReporterDisabled() {
+        String configuration = new KafkaConnectConfigurationBuilder(Reconciliation.DUMMY_RECONCILIATION, BOOTSTRAP_SERVERS)
+                .withStrimziMetricsReporter(null)
+                .build();
+
+        assertThat(configuration, isEquivalent(
+                "admin.security.protocol=PLAINTEXT",
+                "bootstrap.servers=my-cluster-kafka-bootstrap:9092",
+                "security.protocol=PLAINTEXT",
+                "producer.security.protocol=PLAINTEXT",
+                "consumer.security.protocol=PLAINTEXT",
+                StrimziMetricsReporterConfig.LISTENER_ENABLE + "=false",
+                "admin." + StrimziMetricsReporterConfig.LISTENER_ENABLE + "=false",
+                "producer." + StrimziMetricsReporterConfig.LISTENER_ENABLE + "=false",
+                "consumer." + StrimziMetricsReporterConfig.LISTENER_ENABLE + "=false"));
+    }
+
+    static Stream<Arguments> sourceUserConfigWithMetricsReporters() {
+        Map<String, Object> configMap = new HashMap<>();
+        configMap.put("metric.reporters", "my.domain.CustomMetricReporter");
+
+        KafkaConnectConfiguration userConfig = new KafkaConnectConfiguration(Reconciliation.DUMMY_RECONCILIATION, configMap.entrySet());
+        String expectedConfig = "admin.security.protocol=PLAINTEXT\n"
+                + "bootstrap.servers=my-cluster-kafka-bootstrap:9092\n"
+                + "consumer.security.protocol=PLAINTEXT\n"
+                + "config.providers.strimzidir.class=org.apache.kafka.common.config.provider.DirectoryConfigProvider\n"
+                + "config.providers.strimzidir.param.allowed.paths=/opt/kafka\n"
+                + "config.providers.strimzienv.class=org.apache.kafka.common.config.provider.EnvVarConfigProvider\n"
+                + "config.providers.strimzienv.param.allowlist.pattern=.*\n"
+                + "config.providers.strimzifile.class=org.apache.kafka.common.config.provider.FileConfigProvider\n"
+                + "config.providers=strimzienv,strimzifile,strimzidir,strimzisecrets\n"
+                + "config.providers.strimzisecrets.class=io.strimzi.kafka.KubernetesSecretConfigProvider\n"
+                + "config.storage.topic=connect-cluster-configs\n"
+                + "key.converter=org.apache.kafka.connect.json.JsonConverter\n"
+                + "offset.storage.topic=connect-cluster-offsets\n"
+                + "producer.security.protocol=PLAINTEXT\n"
+                + "security.protocol=PLAINTEXT\n"
+                + "status.storage.topic=connect-cluster-status\n"
+                + "group.id=connect-cluster\n"
+                + "value.converter=org.apache.kafka.connect.json.JsonConverter\n";
+
+        // testing 4 combinations of 2 boolean values
+        return Stream.of(
+                Arguments.of(userConfig, false, false,
```

Comment:
The arguments will be much more readable if you split them to lines in the code based on the lines in the config.

Reply:
ok